x-pack/legacy/plugins/snapshot_restore/public/app/components/show_hide_indices.tsx

```diff
@@ -0,0 +1,77 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import React, { useState } from 'react';
+import { EuiDescriptionListDescription, EuiTitle, EuiLink, EuiIcon } from '@elastic/eui';
+interface Props {
+  indices: string[] | string | undefined;
+  defaultState: boolean;
+  i18nId: string;
+}
+
+import { useAppDependencies } from '../index';
+
+export const ShowHideIndices: React.FunctionComponent<Props> = ({
+  indices,
+  defaultState,
+  i18nId,
+}) => {
+  const {
+    core: { i18n },
+  } = useAppDependencies();
+  const { FormattedMessage } = i18n;
+  const [isShowingFullIndicesList, setIsShowingFullIndicesList] = useState<boolean>(defaultState);
+  const displayIndices = indices
+    ? typeof indices === 'string'
+      ? indices.split(',')
+      : indices
+    : undefined;
+  const hiddenIndicesCount =
+    displayIndices && displayIndices.length > 10 ? displayIndices.length - 10 : 0;
+  return (
+    <EuiDescriptionListDescription>
+      {displayIndices ? (
+        <div>
+          <ul>
+            {(isShowingFullIndicesList ? displayIndices : [...displayIndices].splice(0, 10)).map(
+              index => (
+                <li key={index}>
+                  <EuiTitle size="xs">
+                    <span>{index}</span>
+                  </EuiTitle>
+                </li>
+              )
+            )}
+          </ul>
+          {hiddenIndicesCount ? (
+            <EuiLink
+              onClick={() =>
+                isShowingFullIndicesList
+                  ? setIsShowingFullIndicesList(false)
+                  : setIsShowingFullIndicesList(true)
+              }
+            >
+              <FormattedMessage
```

Comment:
A couple suggestions around this:
- I think it's best to have two separate `<FormattedMessage>` components for the "Hide" vs "Show". This makes it easier for whoever ends up translating this. Also, easier for a developer who is reading this code (IMO).
- Since the text is the same, I don't think it's necessary to pass an `i18nId` prop between the components (and might cause issues for translating as it is). Assuming there are two separate `<FormattedMessage>` components as mentioned above, there should just be a common `id` for each, something like `xpack.snapshotRestore.indicesList.indicesCollapseAllLink` and `xpack.snapshotRestore.indicesList.indicesExpandAllLink`

Reply:
Cool. That's a relief because that's easier.